scripts/knot_benchmark/src/benchmark/projects.py

```diff
@@ -0,0 +1,135 @@
+import logging
+import os
+import subprocess
+import typing
+from pathlib import Path
+
+
+class Project(typing.NamedTuple):
+    name: str
+    """The name of the project to benchmark."""
+
+    repository: str
+    """The git repository to clone."""
+
+    revision: str
+
+    dependencies: list[str]
+    """List of type checking dependencies"""
+
+    include: list[str] = []
+    """The directories and files to check. If empty, checks the current directory"""
+
+    pyright_arguments: list[str] | None = None
+    """The arguments passed to pyright. Overrides `include` if set."""
+
+    mypy_arguments: list[str] | None = None
+    """The arguments passed to mypy. Overrides `include` if set."""
+
+    def clone(self, checkout_dir: Path):
+        if os.path.exists(os.path.join(checkout_dir, ".git")):
```

Comment:
```suggestion
        # Don't do anything if `checkout_dir` is a local path 
        if os.path.exists(os.path.join(checkout_dir, ".git")):
```

Reply:
That's not the intention. It's to detect if the project has already been cloned

Reply:
Even more reason to add a comment then, since I clearly misunderstood ;)